integration-tests/src/tests/client/process_blocks.rs

```diff
@@ -2168,7 +2170,25 @@ fn test_validate_chunk_extra() {
         assert_matches!(res.unwrap_err(), near_chain::Error::ChunksMissing(_));
     }
 
-    // Process the previously unavailable chunk. This causes two blocks to be accepted.
+    // Process the previously unavailable chunk. This causes two blocks to be
+    // accepted. Technically, the blocks are accepted concurrently, so we can
+    // observe either `block1 -> block2` reorg or `block2, block1` fork. We want
+    // to try to produce chunks on top of block1, so we force the reorg case
+    // using `capture`
+
+    let barrier = Arc::new(std::sync::Barrier::new(2));
+    capture.set_callback({
+        let block2_hash = *block2.hash();
+        let barrier = Arc::clone(&barrier);
```

Comment:
Relying on logs doesn't sound sustainable :P.
I know we have already talked about this, but have you considered using sth like https://github.com/near/nearcore/blob/ac7c958f7ee239eac25c1130e029d7da2816ae1d/chain/network/src/sink.rs#L10 ?

Reply:
>Relying on logs doesn't sound sustainable :P.

I would actually challenge this -- using logs for this does feel very odd, but I can't think of any specific reason why that'd actually break, especially given that we use structured logging where you can react to a specific event with specific field values (granted, the current impl is string base, but that's just KISS, not a fundamental limitation). 

This *could* work with `Sink`, but:

* that would require threading it through a lot of layers (this is an integration test), and introducing a significant cross-cutting architectural design (so, a risk of half-refactor)
* we already using tracing pervasively, so we'd have essentially two different infrastructures to report events. I can see a benefit (better context tracking primarily) of using something like `Sink<Event>` *instead* of tracing, but well-typed events are a big architectural decision not taken lightly